libraries/AP_Proximity/AP_Proximity_RPLidarA2.cpp

```diff
@@ -165,114 +159,35 @@ void AP_Proximity_RPLidarA2::send_request_for_device_info()
     Debug(1, "Sent device information request");
 }
 
-void AP_Proximity_RPLidarA2::consume_bytes(uint16_t count)
-{
-    if (count > _byte_count) {
-        INTERNAL_ERROR(AP_InternalError::error_t::flow_of_control);
-        _byte_count = 0;
-        return;
-    }
-    _byte_count -= count;
-    if (_byte_count) {
-        memmove((void*)&_payload[0], (void*)&_payload[count], _byte_count);
-    }
-}
-
-void AP_Proximity_RPLidarA2::reset()
-{
-    _state = State::RESET;
-    _byte_count = 0;
-}
-
-bool AP_Proximity_RPLidarA2::make_first_byte_in_payload(uint8_t desired_byte)
-{
-    if (_byte_count == 0) {
-        return false;
-    }
-    if (_payload[0] == desired_byte) {
-        return true;
-    }
-    for (auto i=1; i<_byte_count; i++) {
-        if (_payload[i] == desired_byte) {
-            consume_bytes(i);
-            return true;
-        }
-    }
-    // just not in our buffer.  Throw everything away:
-    _byte_count = 0;
-    return false;
-}
-
 void AP_Proximity_RPLidarA2::get_readings()
 {
-    Debug(2, "             CURRENT STATE: %u ", (unsigned)_state);
-    const uint32_t nbytes = _uart->available();
-    if (nbytes == 0) {
-        return;
-    }
-    const uint32_t bytes_to_read = MIN(nbytes, sizeof(_payload)-_byte_count);
-    if (bytes_to_read == 0) {
-        INTERNAL_ERROR(AP_InternalError::error_t::flow_of_control);
-        reset();
-        return;
-    }
-    const uint32_t bytes_read = _uart->read(&_payload[_byte_count], bytes_to_read);
-    if (bytes_read == 0) {
-        // this is bad; we were told there were bytes available
-        INTERNAL_ERROR(AP_InternalError::error_t::flow_of_control);
-        reset();
-        return;
-    }
-    _byte_count += bytes_read;
-
-    uint32_t previous_loop_byte_count = UINT32_MAX;
-    while (_byte_count) {
-        if (_byte_count >= previous_loop_byte_count) {
-            // this is a serious error, we should always consume some
-            // bytes.  Avoid looping forever.
-            INTERNAL_ERROR(AP_InternalError::error_t::flow_of_control);
-            _uart = nullptr;
-            return;
-        }
-        previous_loop_byte_count = _byte_count;
-
+    Debug(2, "CURRENT STATE: %u ", (unsigned)_state);
+    while (_uart->available()) {
```

Comment:
We normally cap the maximum number of bytes that can be processed in a single update to ensure that a device spamming the UART can't cause the vehicle to crash.  Normally that limit is 1024 bytes but perhaps we need more for this device. Another option might be to add a timeout although the constant checking of time might consume a lot of CPU.

Reply:
@rmackay9 Yes, I will add that, with S2 the driver consume ~400 bytes each update, so a limit of 1024 is not a problem.